packages/ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -609,26 +612,36 @@ export class AngularCompilerPlugin {
     // Exclude the following files from unused checks
     // - ngfactories & ngstyle might not have a correspondent
     //   JS file example `@angular/core/core.ngfactory.ts`.
-    // - .d.ts files might not have a correspondent JS file due to bundling.
     // - __ng_typecheck__.ts will never be requested.
-    const fileExcludeRegExp = /(\.(d|ngfactory|ngstyle|ngsummary)\.ts|ng_typecheck__\.ts)$/;
+    const fileExcludeRegExp = /(\.(ngfactory|ngstyle|ngsummary)\.ts|ng_typecheck__\.ts)$/;
+
+    // Start all the source file names we care about.
+    // Ignore matches to the regexp above, files we've already reported once before, and
+    // node_modules.
+    const sourceFiles = program.getSourceFiles()
```

Comment:
Wouldn't it be simpler to do something like?
```ts
this._typeDeps = newTsProgram
      .getSourceFiles()
      .filter(sf => sf.isDeclarationFile)
      .map(({ fileName }) => fileName);
```

 in `_createOrUpdateProgram` which would also result for this fix to be available for VE compilations

Reply:
Tried adding this at the end of the `_createOrUpdateProgram` method:
```
    console.log(newTsProgram!
      .getSourceFiles()
      .filter(sf => sf.isDeclarationFile && !sf.fileName.includes('node_modules'))
      .map(({ fileName }) => fileName))
```

Then I ran this test. It logged the following:
```
[ 'D:/work/cli/tests/angular_devkit/build_angular/test-project-host-hello-world-app-moqnlwoki9/src/typings.d.ts' ]
[ 'D:/work/cli/tests/angular_devkit/build_angular/test-project-host-hello-world-app-moqnlwoki9/src/typings.d.ts' ]
[ 'D:/work/cli/tests/angular_devkit/build_angular/test-project-host-hello-world-app-moqnlwoki9/src/typings.d.ts' ]
[ 'D:/work/cli/tests/angular_devkit/build_angular/test-project-host-hello-world-app-moqnlwoki9/src/typings.d.ts' ]
[ 'D:/work/cli/tests/angular_devkit/build_angular/test-project-host-hello-world-app-moqnlwoki9/src/typings.d.ts' ]
```

The typings only file was detected, but not the interface files.